packages/frontend/src/app/repository/tests-list/test-details/test-details.component.spec.ts

```diff
@@ -51,4 +68,25 @@ describe('TestDetailsComponent', () => {
       fixture.debugElement.query(By.css('#error-msg-textarea'))
     ).toBeFalsy();
   });
+
+  //test the delete test button
+  it('should redirect to the deleteTest url when delete test button is clicked', fakeAsync(() => {
+    component.orgName = 'testOrg';
+    component.repoName = 'testRepo';
+    component.windowProvider = (mockWindowProvider as unknown) as typeof window;
+    component.comService.fetchDeleteTestUrl = comMock;
+    fixture.detectChanges();
+    //spy on the new window
+    spyOn(component.windowProvider, 'open');
+    //click the test deletion button
+    const deleteButton = fixture.debugElement.nativeElement.querySelector(
+      '#delete-test-button'
+    );
+    deleteButton.click();
+    tick();
+    expect(component.windowProvider.open).toHaveBeenCalledWith(
```

Comment:
neat, I love that the framework we have in place allows for testing behavior like this so easily.

Reply:
Yes, awesome!